<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Quantitative proteomics data imputation — impute • Features</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Quantitative proteomics data imputation — impute" />
<meta property="og:description" content="The impute method performs data imputation on Features and
SummarizedExperiment instance using a variety of methods (see
below).
Users should proceed with care when imputing data and take
precautions to assure that the imputation produce valid results,
in particular with naive imputations such as replacing missing
values with 0." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Features</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Features.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Processing.html">Processing quantitative proteomics data with Features</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RforMassSpectrometry/Features">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Quantitative proteomics data imputation</h1>
    <small class="dont-index">Source: <a href='https://github.com/RforMassSpectrometry/Features/blob/master/R/imputation.R'><code>R/imputation.R</code></a></small>
    <div class="hidden name"><code>impute.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <code>impute</code> method performs data imputation on <code>Features</code> and
<code>SummarizedExperiment</code> instance using a variety of methods (see
below).</p>
<p>Users should proceed with care when imputing data and take
precautions to assure that the imputation produce valid results,
in particular with naive imputations such as replacing missing
values with 0.</p>
    </div>

    <pre class="usage"><span class='no'>impute</span>

<span class='fu'>impute_matrix</span>(<span class='no'>object</span>, <span class='no'>method</span>, <span class='no'>randna</span>, <span class='no'>mar</span>, <span class='no'>mnar</span>, <span class='no'>...</span>)

<span class='co'># S4 method for SummarizedExperiment</span>
<span class='fu'>impute</span>(<span class='no'>object</span>, <span class='no'>method</span>, <span class='no'>randna</span>, <span class='no'>mar</span>, <span class='no'>mnar</span>, <span class='no'>...</span>)

<span class='co'># S4 method for Features</span>
<span class='fu'>impute</span>(<span class='no'>object</span>, <span class='no'>method</span>, <span class='no'>randna</span>, <span class='no'>mar</span>, <span class='no'>mnar</span>, <span class='no'>...</span>, <span class='no'>i</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object with missing values to be imputed.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code> defining the imputation method. See
<code>imputeMethods()</code> for available ones.</p></td>
    </tr>
    <tr>
      <th>randna</th>
      <td><p><code>logical</code> of length equal to <code><a href='https://rdrr.io/r/base/nrow.html'>nrow(object)</a></code> defining
which rows are missing at random. The other ones are
considered missing not at random. Only relevant when <code>methods</code>
is <code>mixed</code>.</p></td>
    </tr>
    <tr>
      <th>mar</th>
      <td><p>Imputation method for values missing at random. See
<code>method</code> above.</p></td>
    </tr>
    <tr>
      <th>mnar</th>
      <td><p>Imputation method for values missing not at
random. See <code>method</code> above.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional parameters passed to the inner imputation
function.</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>Defines which element of the <code>Features</code> instance to
impute. If missing, all elements will be imputed.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>An object of class <code>standardGeneric</code> of length 1.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>There are two types of mechanisms resulting in missing values in
LC/MSMS experiments.</p><ul>
<li><p>Missing values resulting from absence of detection of a feature,
despite ions being present at detectable concentrations. For
example in the case of ion suppression or as a result from the
stochastic, data-dependent nature of the MS acquisition
method. These missing value are expected to be randomly
distributed in the data and are defined as missing at random
(MAR) or missing completely at random (MCAR).</p></li>
<li><p>Biologically relevant missing values resulting from the absence
of the low abundance of ions (below the limit of detection of
the instrument). These missing values are not expected to be
randomly distributed in the data and are defined as missing not
at random (MNAR).</p></li>
</ul>

<p>MNAR features should ideally be imputed with a left-censor method,
such as <code>QRILC</code> below. Conversely, it is recommended to use host
deck methods such nearest neighbours, Bayesian missing value
imputation or maximum likelihood methods when values are missing
at random.</p>
<p>Currently, the following imputation methods are available.</p><ul>
<li><p><em>MLE</em>: Maximum likelihood-based imputation method using the EM
algorithm. Implemented in the <code><a href='https://rdrr.io/pkg/norm/man/imp.norm.html'>norm::imp.norm()</a></code>. function. See
<code><a href='https://rdrr.io/pkg/norm/man/imp.norm.html'>norm::imp.norm()</a></code> for details and additional parameters. Note
that here, <code>...</code> are passed to the [norm::em.norm()<code>function, rather to the actual imputation function</code>imp.norm`.</p></li>
<li><p><em>bpca</em>: Bayesian missing value imputation are available, as
implemented in the <code><a href='https://rdrr.io/pkg/pcaMethods/man/pca.html'>pcaMethods::pca()</a></code> function. See
<code><a href='https://rdrr.io/pkg/pcaMethods/man/pca.html'>pcaMethods::pca()</a></code> for details and additional parameters.</p></li>
<li><p><em>knn</em>: Nearest neighbour averaging, as implemented in the
<code><a href='https://rdrr.io/pkg/impute/man/impute.knn.html'>impute::impute.knn</a></code> function. See <code><a href='https://rdrr.io/pkg/impute/man/impute.knn.html'>impute::impute.knn()</a></code>] for
details and additional parameters.</p></li>
<li><p><em>QRILC</em>: A missing data imputation method that performs the
imputation of left-censored missing data using random draws from
a truncated distribution with parameters estimated using
quantile regression. Implemented in the
<code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.QRILC.html'>imputeLCMD::impute.QRILC</a></code>
function. <code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.QRILC.html'>imputeLCMD::impute.QRILC()</a></code> for details and
additional parameters.</p></li>
<li><p><em>MinDet</em>: Performs the imputation of left-censored missing data
using a deterministic minimal value approach. Considering a
expression data with <em>n</em> samples and <em>p</em> features, for each
sample, the missing entries are replaced with a minimal value
observed in that sample. The minimal value observed is estimated
as being the q-th quantile (default <code>q = 0.01</code>) of the observed
values in that sample. Implemented in the
<code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.MinDet.html'>imputeLCMD::impute.MinDet</a></code> function. See
<code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.MinDet.html'>imputeLCMD::impute.MinDet()</a></code> for details and additional
parameters.</p></li>
<li><p><em>MinProb</em>: Performs the imputation of left-censored missing data
by random draws from a Gaussian distribution centred to a
minimal value. Considering an expression data matrix with <em>n</em>
samples and <em>p</em> features, for each sample, the mean value of the
Gaussian distribution is set to a minimal observed value in that
sample. The minimal value observed is estimated as being the
q-th quantile (default <code>q = 0.01</code>) of the observed values in
that sample. The standard deviation is estimated as the median
of the feature standard deviations. Note that when estimating
the standard deviation of the Gaussian distribution, only the
peptides/proteins which present more than 50\<!-- % recorded values -->
are considered. Implemented in the <code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.MinProb.html'>imputeLCMD::impute.MinProb</a></code>
function. See <code><a href='https://rdrr.io/pkg/imputeLCMD/man/impute.MinProb.html'>imputeLCMD::impute.MinProb()</a></code> for details and
additional parameters.</p></li>
<li><p><em>min</em>: Replaces the missing values by the smallest non-missing
value in the data.</p></li>
<li><p><em>zero</em>: Replaces the missing values by 0. See also <code><a href='Features-missing-data.html'>zeroIsNA()</a></code>.</p></li>
<li><p><em>mixed</em>: A mixed imputation applying two methods (to be defined
by the user as <code>mar</code> for values missing at random and <code>mnar</code> for
values missing not at random, see example) on two M<a href='https://rdrr.io/r/stats/zC.html'>C</a>AR/MNAR
subsets of the data (as defined by the user by a <code>randna</code>
logical, of length equal to nrow(object)).</p></li>
<li><p><em>nbavg</em>: Average neighbour imputation for fractions collected
along a fractionation/separation gradient, such as sub-cellular
fractions. The method assumes that the fraction are ordered
along the gradient and is invalid otherwise.</p>
<p>Continuous sets <code>NA</code> value at the beginning and the end of the
quantitation vectors are set to the lowest observed value in the
data or to a user defined value passed as argument <code>k</code>. Them,
when a missing value is flanked by two non-missing neighbouring
values, it is imputed by the mean of its direct neighbours. A
stretch of 2 or more missing values will not be imputed. See the
example below.</p></li>
<li><p><em>none</em>: No imputation is performed and the missing values are
left untouched. Implemented in case one wants to only impute
value missing at random or not at random with the <em>mixed</em>
method.</p></li>
</ul>

<p>The <code>imputeMethods()</code> function returns a vector with valid
imputation method arguments.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Olga Troyanskaya, Michael Cantor, Gavin Sherlock, Pat Brown,
Trevor Hastie, Robert Tibshirani, David Botstein and Russ B.
Altman, Missing value estimation methods for DNA microarrays
Bioinformatics (2001) 17 (6): 520-525.</p>
<p>Oba et al., A Bayesian missing value estimation method for gene
expression profile data, Bioinformatics (2003) 19 (16): 2088-2096.</p>
<p>Cosmin Lazar (2015). imputeLCMD: A collection of methods for
left-censored missing data imputation. R package version
2.0. <a href='http://CRAN.R-project.org/package=imputeLCMD'>http://CRAN.R-project.org/package=imputeLCMD</a>.</p>
<p>Lazar C, Gatto L, Ferro M, Bruley C, Burger T. Accounting for the
Multiple Natures of Missing Values in Label-Free Quantitative
Proteomics Data Sets to Compare Imputation Strategies. J Proteome
Res. 2016 Apr 1;15(4):1116-25. doi:
10.1021/acs.jproteome.5b00981. PubMed PMID:26906401.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>imputeMethods</span>()</div><div class='output co'>#&gt;  [1] "bpca"    "knn"     "QRILC"   "MLE"     "MinDet"  "MinProb" "min"    
#&gt;  [8] "zero"    "mixed"   "nbavg"   "none"   </div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>se_na2</span>)
<span class='co'>## table of missing values along the rows (proteins)</span>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span>(<span class='fu'>rowData</span>(<span class='no'>se_na2</span>)$<span class='no'>nNA</span>)</div><div class='output co'>#&gt; 
#&gt;   0   1   2   3   4   8   9  10 
#&gt; 301 247  91  13   2  23  10   2 </div><div class='input'><span class='co'>## table of missing values along the columns (samples)</span>
<span class='fu'>colData</span>(<span class='no'>se_na2</span>)$<span class='no'>nNA</span></div><div class='output co'>#&gt;  [1] 34 45 56 39 47 52 49 61 41 42 55 45 51 43 57 53</div><div class='input'>
<span class='co'>## non-random missing values</span>
<span class='no'>notna</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span>(!<span class='fu'>rowData</span>(<span class='no'>se_na2</span>)$<span class='no'>randna</span>)
<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>notna</span>)</div><div class='output co'>#&gt; [1] 35</div><div class='input'><span class='no'>notna</span></div><div class='output co'>#&gt;  [1]   6  20  79  88 130 187 227 231 238 264 275 317 324 363 373 382 409 437 445
#&gt; [20] 453 456 474 484 485 492 514 516 546 568 580 594 631 648 664 671</div><div class='input'>
<span class='fu'>impute</span>(<span class='no'>se_na2</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"min"</span>)</div><div class='output co'>#&gt; class: SummarizedExperiment 
#&gt; dim: 689 16 
#&gt; metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion
#&gt; assays(1): ''
#&gt; rownames(689): AT1G09210 AT1G21750 ... AT4G11150 AT4G39080
#&gt; rowData names(2): nNA randna
#&gt; colnames(16): M1F1A M1F4A ... M2F8B M2F11B
#&gt; colData names(1): nNA</div><div class='input'>
<span class='kw'>if</span> (<span class='fu'><a href='https://rdrr.io/r/base/library.html'>require</a></span>(<span class='st'>"imputeLCMD"</span>)) {
  <span class='fu'>impute</span>(<span class='no'>se_na2</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"QRILC"</span>)
  <span class='fu'>impute</span>(<span class='no'>se_na2</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"MinDet"</span>)
}</div><div class='output co'>#&gt; <span class='message'>Loading required package: imputeLCMD</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: tmvtnorm</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: mvtnorm</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Matrix</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Matrix’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:S4Vectors’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: gmm</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: sandwich</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: norm</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: pcaMethods</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘pcaMethods’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    loadings</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: impute</span></div><div class='output co'>#&gt; class: SummarizedExperiment 
#&gt; dim: 689 16 
#&gt; metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion
#&gt; assays(1): ''
#&gt; rownames(689): AT1G09210 AT1G21750 ... AT4G11150 AT4G39080
#&gt; rowData names(2): nNA randna
#&gt; colnames(16): M1F1A M1F4A ... M2F8B M2F11B
#&gt; colData names(1): nNA</div><div class='input'>
<span class='kw'>if</span> (<span class='fu'><a href='https://rdrr.io/r/base/library.html'>require</a></span>(<span class='st'>"norm"</span>))
  <span class='fu'>impute</span>(<span class='no'>se_na2</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"MLE"</span>)</div><div class='output co'>#&gt; Iterations of EM: 
#&gt; 1...2...3...4...5...6...7...8...9...10...11...</div><div class='output co'>#&gt; class: SummarizedExperiment 
#&gt; dim: 689 16 
#&gt; metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion
#&gt; assays(1): ''
#&gt; rownames(689): AT1G09210 AT1G21750 ... AT4G11150 AT4G39080
#&gt; rowData names(2): nNA randna
#&gt; colnames(16): M1F1A M1F4A ... M2F8B M2F11B
#&gt; colData names(1): nNA</div><div class='input'>
<span class='fu'>impute</span>(<span class='no'>se_na2</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"mixed"</span>,
       <span class='kw'>randna</span> <span class='kw'>=</span> <span class='fu'>rowData</span>(<span class='no'>se_na2</span>)$<span class='no'>randna</span>,
       <span class='kw'>mar</span> <span class='kw'>=</span> <span class='st'>"knn"</span>, <span class='kw'>mnar</span> <span class='kw'>=</span> <span class='st'>"QRILC"</span>)</div><div class='output co'>#&gt; class: SummarizedExperiment 
#&gt; dim: 689 16 
#&gt; metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion
#&gt; assays(1): ''
#&gt; rownames(689): AT1G09210 AT1G21750 ... AT4G11150 AT4G39080
#&gt; rowData names(2): nNA randna
#&gt; colnames(16): M1F1A M1F4A ... M2F8B M2F11B
#&gt; colData names(1): nNA</div><div class='input'>
<span class='co'>## neighbour averaging</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='no'>se_na2</span>[<span class='fl'>1</span>:<span class='fl'>4</span>, <span class='fl'>1</span>:<span class='fl'>6</span>]
<span class='fu'>assay</span>(<span class='no'>x</span>)[<span class='fl'>1</span>, <span class='fl'>1</span>] <span class='kw'>&lt;-</span> <span class='fl'>NA</span> <span class='co'>## min value</span>
<span class='fu'>assay</span>(<span class='no'>x</span>)[<span class='fl'>2</span>, <span class='fl'>3</span>] <span class='kw'>&lt;-</span> <span class='fl'>NA</span> <span class='co'>## average</span>
<span class='fu'>assay</span>(<span class='no'>x</span>)[<span class='fl'>3</span>, <span class='fl'>1</span>:<span class='fl'>2</span>] <span class='kw'>&lt;-</span> <span class='fl'>NA</span> <span class='co'>## min value and average</span>
<span class='co'>## 4th row: no imputation</span>
<span class='fu'>assay</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt;              M1F1A    M1F4A   M1F7A  M1F11A    M1F2B    M1F5B
#&gt; AT1G09210       NA 0.275500 0.21600 0.18525 0.465667 0.199667
#&gt; AT1G21750 0.332000 0.279667      NA 0.16600 0.451500 0.200375
#&gt; AT1G51760       NA       NA 0.16825 0.18825 0.459750 0.214500
#&gt; AT1G56340 0.336733       NA      NA      NA 0.487167 0.201833</div><div class='input'>
<span class='fu'>assay</span>(<span class='fu'>impute</span>(<span class='no'>x</span>, <span class='st'>"nbavg"</span>))</div><div class='output co'>#&gt; <span class='message'>Assuming values are ordered.</span></div><div class='output co'>#&gt;              M1F1A    M1F4A     M1F7A  M1F11A    M1F2B    M1F5B
#&gt; AT1G09210 0.166000 0.275500 0.2160000 0.18525 0.465667 0.199667
#&gt; AT1G21750 0.332000 0.279667 0.2228335 0.16600 0.451500 0.200375
#&gt; AT1G51760 0.166000 0.167125 0.1682500 0.18825 0.459750 0.214500
#&gt; AT1G56340 0.336733       NA        NA      NA 0.487167 0.201833</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#format">Format</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#references">References</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


