<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing quantitative proteomics data with Features â€¢ Features</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing quantitative proteomics data with Features">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Features</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Features.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Processing.html">Processing quantitative proteomics data with Features</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RforMassSpectrometry/Features">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Processing quantitative proteomics data with Features</h1>
                        <h4 class="author">Laurent Gatto</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/RforMassSpectrometry/Features/blob/master/vignettes/Processing.Rmd"><code>vignettes/Processing.Rmd</code></a></small>
      <div class="hidden name"><code>Processing.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      This vignette describes how to process quantitative mass spectrometry data with Features: cleaning up unneeded feature variables, adding an experimental design, filtering out contaminants and reverse hits, managing missing values, log-transforming, normalising and aggregating data.
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="reading-data-as-features" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-data-as-features" class="anchor"></a>Reading data as <code>Features</code>
</h2>
<p>We are going to use a subset of the CPTAC study 6 containing conditions A and B. The peptide-level data, as processed by MaxQuant is available in the <code>msdata</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(f &lt;-<span class="st"> </span>msdata<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/msdata/man/proteomics.html">quant</a></span>(<span class="dt">pattern =</span> <span class="st">"cptac"</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</a></code></pre></div>
<pre><code>## [1] "cptac_a_b_peptides.txt"</code></pre>
<p>From the names of the columns, we see that the quantitative columns, starting with <code>"Intensity."</code> (note the space!) are at positions 56 to 61.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(f))</a></code></pre></div>
<pre><code>##  [1] "Sequence"                 "N.term.cleavage.window"  
##  [3] "C.term.cleavage.window"   "Amino.acid.before"       
##  [5] "First.amino.acid"         "Second.amino.acid"       
##  [7] "Second.last.amino.acid"   "Last.amino.acid"         
##  [9] "Amino.acid.after"         "A.Count"                 
## [11] "R.Count"                  "N.Count"                 
## [13] "D.Count"                  "C.Count"                 
## [15] "Q.Count"                  "E.Count"                 
## [17] "G.Count"                  "H.Count"                 
## [19] "I.Count"                  "L.Count"                 
## [21] "K.Count"                  "M.Count"                 
## [23] "F.Count"                  "P.Count"                 
## [25] "S.Count"                  "T.Count"                 
## [27] "W.Count"                  "Y.Count"                 
## [29] "V.Count"                  "U.Count"                 
## [31] "Length"                   "Missed.cleavages"        
## [33] "Mass"                     "Proteins"                
## [35] "Leading.razor.protein"    "Start.position"          
## [37] "End.position"             "Unique..Groups."         
## [39] "Unique..Proteins."        "Charges"                 
## [41] "PEP"                      "Score"                   
## [43] "Identification.type.6A_7" "Identification.type.6A_8"
## [45] "Identification.type.6A_9" "Identification.type.6B_7"
## [47] "Identification.type.6B_8" "Identification.type.6B_9"
## [49] "Experiment.6A_7"          "Experiment.6A_8"         
## [51] "Experiment.6A_9"          "Experiment.6B_7"         
## [53] "Experiment.6B_8"          "Experiment.6B_9"         
## [55] "Intensity"                "Intensity.6A_7"          
## [57] "Intensity.6A_8"           "Intensity.6A_9"          
## [59] "Intensity.6B_7"           "Intensity.6B_8"          
## [61] "Intensity.6B_9"           "Reverse"                 
## [63] "Potential.contaminant"    "id"                      
## [65] "Protein.group.IDs"        "Mod..peptide.IDs"        
## [67] "Evidence.IDs"             "MS.MS.IDs"               
## [69] "Best.MS.MS"               "Oxidation..M..site.IDs"  
## [71] "MS.MS.Count"</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">(i &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"Intensity</span><span class="ch">\\</span><span class="st">."</span>, <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(f))))</a></code></pre></div>
<pre><code>## [1] 56 57 58 59 60 61</code></pre>
<p>We now read these data using the <code>readFeatures</code> function. The peptide level expression data will be imported into R as an instance of class <code>Features</code> named <code>cptac</code> with an assay named <code>peptides</code>. We also use the <code>fnames</code> argument to set the row-names of the <code>peptides</code> assay to the peptide sequences.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"Features"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/readFeatures.html">readFeatures</a></span>(f, <span class="dt">ecol =</span> i, <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, <span class="dt">name =</span> <span class="st">"peptides"</span>, <span class="dt">fnames =</span> <span class="st">"Sequence"</span>)</a></code></pre></div>
</div>
<div id="encoding-the-experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#encoding-the-experimental-design" class="anchor"></a>Encoding the experimental design</h2>
<p>Below we update the sample (column) annotations to encode the two groups, 6A and 6B, and the original sample numbers.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">cptac<span class="op">$</span>group &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"6A"</span>, <span class="st">"6B"</span>), <span class="dt">each =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">cptac<span class="op">$</span>sample &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="dv">7</span><span class="op">:</span><span class="dv">9</span>, <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">colData</span>(cptac)</a></code></pre></div>
<pre><code>## DataFrame with 6 rows and 2 columns
##                      group    sample
##                &lt;character&gt; &lt;integer&gt;
## Intensity.6A_7          6A         7
## Intensity.6A_8          6A         8
## Intensity.6A_9          6A         9
## Intensity.6B_7          6B         7
## Intensity.6B_8          6B         8
## Intensity.6B_9          6B         9</code></pre>
</div>
<div id="filtering-out-contaminants-and-reverse-hits" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-out-contaminants-and-reverse-hits" class="anchor"></a>Filtering out contaminants and reverse hits</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="../reference/Features-filtering.html">filterFeatures</a></span>(cptac, <span class="op">~</span><span class="st"> </span>Reverse <span class="op">==</span><span class="st"> ""</span>)</a></code></pre></div>
<pre><code>## A instance of class Features containing
##  [1] peptides: SummarizedExperiment with 11436 rows and 6 columns</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="../reference/Features-filtering.html">filterFeatures</a></span>(cptac, <span class="op">~</span><span class="st"> </span>Potential.contaminant <span class="op">==</span><span class="st"> ""</span>)</a></code></pre></div>
<pre><code>## A instance of class Features containing
##  [1] peptides: SummarizedExperiment with 11385 rows and 6 columns</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"magrittr"</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">cptac &lt;-<span class="st"> </span>cptac <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="../reference/Features-filtering.html">filterFeatures</a></span>(<span class="op">~</span><span class="st"> </span>Reverse <span class="op">==</span><span class="st"> ""</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="../reference/Features-filtering.html">filterFeatures</a></span>(<span class="op">~</span><span class="st"> </span>Potential.contaminant <span class="op">==</span><span class="st"> ""</span>)</a></code></pre></div>
</div>
<div id="removing-up-unneeded-feature-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#removing-up-unneeded-feature-variables" class="anchor"></a>Removing up unneeded feature variables</h2>
<p>The spreadsheet that was read above contained numerous variables that are returned by MaxQuant, but not necessarily necessary in the frame of a downstream statistical analysis.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/Features-class.html">rowDataNames</a></span>(cptac)</a></code></pre></div>
<pre><code>## CharacterList of length 1
## [["peptides"]] Sequence N.term.cleavage.window ... MS.MS.Count</code></pre>
<p>The only ones that we will be needing below are the peptides sequences and the protein identifiers. Below, we store these variables of interest and filter them using the <code>selectRowData</code> function.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">rowvars &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Sequence"</span>, <span class="st">"Proteins"</span>, <span class="st">"Leading.razor.protein"</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Features-class.html">selectRowData</a></span>(cptac, rowvars)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw"><a href="../reference/Features-class.html">rowDataNames</a></span>(cptac)</a></code></pre></div>
<pre><code>## CharacterList of length 1
## [["peptides"]] Sequence Proteins Leading.razor.protein</code></pre>
</div>
<div id="managing-missing-values" class="section level2">
<h2 class="hasAnchor">
<a href="#managing-missing-values" class="anchor"></a>Managing missing values</h2>
<p>Missing values can be very numerous in certain proteomics experiments and need to be dealt with carefully. The first step is to assess their presence across samples and features. But before being able to do so, we need to replace 0 by <code>NA</code>, given that MaxQuant encodes missing data with a 0 using the <code>zeroIsNA</code> function.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Features-missing-data.html">zeroIsNA</a></span>(cptac)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw"><a href="../reference/Features-missing-data.html">nNA</a></span>(cptac)</a></code></pre></div>
<pre><code>## $nNA
## [1] 0.4491943
## 
## $nNArows
## 
##    0    1    2    3    4    5    6 
## 4051  989  879  715  912  797 3014 
## 
## $nNAcols
## Intensity.6A_7 Intensity.6A_8 Intensity.6A_9 Intensity.6B_7 Intensity.6B_8 
##           4669           5388           5224           4651           5470 
## Intensity.6B_9 
##           5207</code></pre>
<p>The output of the <code>nNA</code> function tells us that</p>
<ul>
<li>there are currently close to 50% is missing values in the data;</li>
<li>there are 4051 peptides with 0 missing values, 989 with a single missing values, â€¦ and 3014 peptides composed of only missing values;</li>
<li>the range of missing values in the 6 samples is comparable and ranges between 4651 and 5470.</li>
</ul>
<p>In this dataset, we have such a high number of peptides without any data because the 6 samples are a subset of a larger dataset, and these peptides happened to be absent in groups A and B. Below, we use <code>filterNA</code> to remove these peptides. By setting <code>pNA</code> to 0.9, we allow peptides that have up to 90% of missing values, including in this case those that have only one value (i.e 5/6 percent of missing values).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Features-missing-data.html">filterNA</a></span>(cptac, <span class="dt">pNA =</span> <span class="fl">.9</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">cptac</a></code></pre></div>
<pre><code>## A instance of class Features containing
##  [1] peptides: SummarizedExperiment with 8343 rows and 6 columns</code></pre>
<p>If we wanted to remove all peptides that have one or more missing values, we could simply use the default value of <code>pNA = 0</code>.</p>
</div>
<div id="data-transformation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-transformation" class="anchor"></a>Data transformation</h2>
<p>When analysing continuous data using parametric methods (such as t-test or linear models), it is often necessary to log-transform the data. The figure below (left) show that how our data is mainly composed of small values with a long tail of larger ones, which is a typical pattern of quantitative omics data.</p>
<p>Below, we use the <code>logTransform</code> function to log2-transform our data. This time, instead of overwriting the peptides assay, we are going to create a new one to contain the log2-transformed data.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Features-class.html">addAssay</a></span>(cptac,</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">                  <span class="kw"><a href="../reference/Features-processing.html">logTransform</a></span>(cptac[[<span class="dv">1</span>]]),</a>
<a class="sourceLine" id="cb23-3" data-line-number="3">                  <span class="dt">name =</span> <span class="st">"peptides_log"</span>)</a>
<a class="sourceLine" id="cb23-4" data-line-number="4">cptac</a></code></pre></div>
<pre><code>## A instance of class Features containing
##  [1] peptides: SummarizedExperiment with 8343 rows and 6 columns 
##  [2] peptides_log: SummarizedExperiment with 8343 rows and 6 columns</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="dt">mfrow =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">limma<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html">plotDensities</a></span>(<span class="kw">assay</span>(cptac[[<span class="dv">1</span>]]))</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">limma<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html">plotDensities</a></span>(<span class="kw">assay</span>(cptac[[<span class="dv">2</span>]]))</a></code></pre></div>
<div class="figure">
<img src="Processing_files/figure-html/unnamed-chunk-10-1.png" alt="Quantitative data in its original scale (left) and log2-transformed (right)." width="700"><p class="caption">
Quantitative data in its original scale (left) and log2-transformed (right).
</p>
</div>
</div>
<div id="normalisation" class="section level2">
<h2 class="hasAnchor">
<a href="#normalisation" class="anchor"></a>Normalisation</h2>
<p>Assays in <code>Features</code> objects can be normalised with the <code>normalize</code> function. The type of normalisation is defined by the <code>method</code> argument; below, we use quantiles normalisation, store the normalised data into a new experiment, and visualise the resulting data.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">cptac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/Features-class.html">addAssay</a></span>(cptac,</a>
<a class="sourceLine" id="cb26-2" data-line-number="2">                  <span class="kw"><a href="../reference/Features-processing.html">normalize</a></span>(cptac[[<span class="st">"peptides_log"</span>]], <span class="dt">method =</span> <span class="st">"quantiles"</span>),</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">                  <span class="dt">name =</span> <span class="st">"peptides_norm"</span>)</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">cptac</a></code></pre></div>
<pre><code>## A instance of class Features containing
##  [1] peptides: SummarizedExperiment with 8343 rows and 6 columns 
##  [2] peptides_log: SummarizedExperiment with 8343 rows and 6 columns 
##  [3] peptides_norm: SummarizedExperiment with 8343 rows and 6 columns</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="dt">mfrow =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">limma<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html">plotDensities</a></span>(<span class="kw">assay</span>(cptac[[<span class="st">"peptides_log"</span>]]))</a>
<a class="sourceLine" id="cb28-3" data-line-number="3">limma<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/plotDensities.html">plotDensities</a></span>(<span class="kw">assay</span>(cptac[[<span class="st">"peptides_norm"</span>]]))</a></code></pre></div>
<div class="figure">
<img src="Processing_files/figure-html/unnamed-chunk-12-1.png" alt="Distribution of log2 peptide intensities before (left) and after (right) quantiles normalisation." width="700"><p class="caption">
Distribution of log2 peptide intensities before (left) and after (right) quantiles normalisation.
</p>
</div>
</div>
<div id="feature-aggregation" class="section level2">
<h2 class="hasAnchor">
<a href="#feature-aggregation" class="anchor"></a>Feature aggregation</h2>
<p>At this stage, it is possible to directly use the peptide-level intensities to perform a statistical analysis, or aggregate the peptide-level data into protein intensities, and perform the differential expression analysis at the protein level.</p>
<p>To aggregate feature data, we can use the <code>aggregateFeature</code> function that takes the following inputs:</p>
</div>
<div id="todo" class="section level2">
<h2 class="hasAnchor">
<a href="#todo" class="anchor"></a>TODO</h2>
<ul>
<li>visualisation</li>
<li>imputation</li>
</ul>
</div>
</div>
<div id="session-information" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R Under development (unstable) (2019-12-13 r77566)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] magrittr_1.5                Features_0.2.0             
##  [3] MultiAssayExperiment_1.13.3 SummarizedExperiment_1.17.0
##  [5] DelayedArray_0.13.0         BiocParallel_1.21.0        
##  [7] matrixStats_0.55.0          Biobase_2.47.1             
##  [9] GenomicRanges_1.39.1        GenomeInfoDb_1.23.0        
## [11] IRanges_2.21.2              S4Vectors_0.25.6           
## [13] BiocGenerics_0.33.0         BiocStyle_2.15.2           
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.3              msdata_0.27.0           highr_0.8              
##  [4] compiler_4.0.0          BiocManager_1.30.10     XVector_0.27.0         
##  [7] ProtGenerics_1.19.2     bitops_1.0-6            tools_4.0.0            
## [10] zlibbioc_1.33.0         digest_0.6.23           preprocessCore_1.49.0  
## [13] lattice_0.20-38         evaluate_0.14           memoise_1.1.0          
## [16] rlang_0.4.2             Matrix_1.2-18           yaml_2.2.0             
## [19] pkgdown_1.4.1           xfun_0.11               GenomeInfoDbData_1.2.2 
## [22] stringr_1.4.0           knitr_1.26              MsCoreUtils_0.0.4      
## [25] desc_1.2.0              fs_1.3.1                grid_4.0.0             
## [28] rprojroot_1.3-2         R6_2.4.1                rmarkdown_2.0          
## [31] bookdown_0.16           limma_3.43.0            AnnotationFilter_1.11.0
## [34] backports_1.1.5         htmltools_0.4.0         MASS_7.3-51.4          
## [37] assertthat_0.2.1        stringi_1.4.3           lazyeval_0.2.2         
## [40] RCurl_1.95-4.12         crayon_1.3.4</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#reading-data-as-features">Reading data as <code>Features</code></a></li>
      <li><a href="#encoding-the-experimental-design">Encoding the experimental design</a></li>
      <li><a href="#filtering-out-contaminants-and-reverse-hits">Filtering out contaminants and reverse hits</a></li>
      <li><a href="#removing-up-unneeded-feature-variables">Removing up unneeded feature variables</a></li>
      <li><a href="#managing-missing-values">Managing missing values</a></li>
      <li><a href="#data-transformation">Data transformation</a></li>
      <li><a href="#normalisation">Normalisation</a></li>
      <li><a href="#feature-aggregation">Feature aggregation</a></li>
      <li><a href="#todo">TODO</a></li>
      </ul>
</li>
      <li><a href="#session-information">Session information</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
